Normative: Add baseName getter

littledan opened the pull request:
The "base name" of a locale stands for the language-region-script
triple. This patch adds a baseName getter on Intl.Locale.prototype
to allow for easy access to this part of the locale.

Closes #22

littledan committed:
Normative: Add baseName getter

The "base name" of a locale stands for the language-region-script
triple. This patch adds a baseName getter on Intl.Locale.prototype
to allow for easy access to this part of the locale.

Closes #22

littledan commented:
cc @nciric

zbraniecki commented:
What about variants and privatetags? I thought we decided to include them in the baseName. I don't know if there's any standardized model for `baseName` that we could reuse which could tell us what subtags should go in.

littledan commented:
@zbraniecki Oh that's right, not sure how I forgot. Will fix.

littledan commented:
Hmm, was the idea to leave in `privateuse` all the time, or only when it's a `privateuse` language tag entirely? The notes are a bit short here compared to my memory of the discussion.

littledan committed:
Normative: Fix baseName definition

- Return the whole tag for entirely private use/grandfathered
- Include the variants on langtag
- Continue to omit extensions, including private use extensions

anba commented:
Do want to handle grandfathered tags like in #24 and return `undefined` for them or do we want to support them for `baseName`?

Nit: The `"baseName"` accessor should be moved before `"calendar"`.

littledan commented:
@anba I thought we should return the whole locale in this case. That's what this patch does. Does anyone else have other opinions here?

anba commented:
Meh, I've read it as `return undefined` for the "does not match langtag" case, similar to the other getters. So, just ignore my comment and carry on! :-)

littledan commented:
@anba I think the difference here is, `language` gets at something specific, which you can't extract from those funny tags, whereas `baseName` is more about, "strip off all the extensions".
